SzczepionkaAPI/src/main/java/com/szczepionka/service/AppointmentService.java

```diff
@@ -64,4 +66,35 @@ private int getRandomMinutesOffset() {
     private LocationDetails getLocationDetailsById(long id) {
         return modelMapper.map(vaccinationLocationsFetcher.getById(id), LocationDetails.class);
     }
+
+    public Appointment cancelAppointment(Long appointmentId) {
+        Optional<Appointment> appointment = appointmentRepository.findById(appointmentId);
+        if(appointment.isPresent()) {
+            appointment.get().setAppointmentStatus(AppointmentStatus.CANCELLED);
+            return appointmentRepository.save(appointment.get());
+        }
+        return null;
+    }
+
+    public AppointmentDetailsDTO getAppointmentDetails(Long appointmentId) {
+        Optional<Appointment> appointment = appointmentRepository.findById(appointmentId);
+        if(appointment.isPresent()) {
+            LocationDetails locationDetails = appointment.get().getLocationDetails();
+            Patient patient = patientService.findPatient(appointment.get().getPatientId());
+
+            return AppointmentDetailsDTO.builder()
```

Comment:
Maybe consider using mapper?

Reply:
those fields are set based on few different places (some from the patient, some from appointment), not sure how to implement a mapper here